tutorials/continuous-deployment-pipeline-google-container-engine-with-codeship/index.md

```diff
@@ -0,0 +1,275 @@
+to `.env`.
+
+Inside the file, you will replace the `...` with your Google Cloud Platform
+project details as follows:
+
+1.  `DEFAULT_ZONE`: Select a default [compute zone](https://cloud.google.com/compute/docs/regions-zones/regions-zones#available)
+    like `us-central1-b`
+1.  `APP_NAME`: Give your application a name, in this case use `hello-express`
+1.  `CONTAINER_CLUSTER`: Give your container cluster a name, in this case use
+    `hello-express-cluster`
+1.  `GOOGLE_PROJECT_ID`: The `Project ID` is found in the [Google Cloud Engine dashboard](https://console.cloud.google.com/home/dashboard)
+1.  `GOOGLE_AUTH_EMAIL`: Use the `Service Account ID` for the [service account](https://console.cloud.google.com/iam-admin/serviceaccounts)
+    created in the last step.
+1.  `GOOGLE_AUTH_JSON`: Add the credentials downloaded in the previous step
+    here. You must replace the newlines with spaces by running
+    `tr '\n' ' ' < your_file_name`.
+
+Save the `.env` file once these items are finished.
+
+### Step 3: Run initial deployment
+
+You need to push an image and create your clusters in GKE initially before you
+can set up a fully automated pipeline in Codeship. In the `hello-express` source
+code is a `bin` folder with `bash` scripts perform these tasks using the
+Codeship Jet CLI.
+
+1.  Make sure the scripts in the `bin` folder are executable. In a terminal
+    window, navigate to the project folder and run the following command:
+
+        chmod +x ./bin/**
+
+1.  Update the `image` line in the `codeship-service.yml` file with your Google
+    Cloud Platform project ID.
+
+        app:
+          build:
+            dockerfile: Dockerfile
+            image: gcr.io/YOUR_PROJECT_IDE/hello-express #update this line using your Google Cloud Platform project ID
+        ...
+
+1.  Update the `image_name` line in the `codeship_steps.yml` file with your
+    Google Cloud Platform project ID.
+
+        - name: build-image
+          service: app
+          command: echo "Build completed"
+        - name: push-image-with-sha
+          service: app
+          type: push
+          image_name: "gcr.io/YOUR_PROJECT_ID/hello-express" #update this line using your Google Cloud Platform project ID
+          image_tag: "{{printf \"%.8s\" .CommitID}}"
+          registry: https://gcr.io
+          dockercfg_service: codeship_gcr_dockercfg
+        - name: tag-as-master
+          service: app
+          type: push
+          tag: master
+          image_name: "gcr.io/YOUR_PROJECT_ID/hello-express" #update this line using your Google Cloud Platform project ID
+          image_tag: "master"
+          registry: https://gcr.io
+          dockercfg_service: codeship_gcr_dockercfg
+        - name: gke-initial-deployment
+          service: codeship_gce_service
+          tag: master
+          command: bin/deploy --create
+        ...
+
+This pipeline will build the Docker image, and then push to Google Container
```

Comment:
This sentence is confusing, and I'm not sure I understand it well enough to propose a rewritten version. Please clarify. (It might help to break this up into multiple sentences.)

Reply:
Here is a complete rewrite of this section for clarity. Let me know if this is an improvement, or adds to the confusion.

> 
> This pipeline runs each step in series. The `build-image` step instructs Codeship to build the `hello-express` Docker image on the CI server. After Codeship builds the Docker image, the `push-image-with-sha` step will push the image to Google Container Registry using the name `gcr.io/YOUR_PROJECT_ID/hello-express`, adding a tag using the first 8 characters of the commit SHA for every commit to the repository. 
> 
> The third and fourth step will run only if the branch is tagged as `master`. The `tag-as-master` step will add the `master` tag to the image pushed to Google Container Registry. This indicates the image in Google Container Registry that is currently deployed. The following step, `gke-initial-deployment`, builds the Container Engine cluster and deploys the `gcr.io/YOUR_PROJECT_ID/hello-express` Docker image.